tests/unittests/algo/test_tpe.py

```diff
@@ -316,6 +424,69 @@ def test_split_trials(self, tpe):
         assert below_points == [[-3.0], [-2.4]]
         assert len(above_points) == 8
 
+    def test_sample_int_dimension(self):
+        """Test sample values for a integer dimension"""
+        space = Space()
+        dim1 = Integer('yolo1', 'uniform', -10, 20)
+        space.register(dim1)
+
+        dim2 = Integer('yolo2', 'uniform', -5, 10, shape=(2))
+        space.register(dim2)
+
+        tpe = TPE(space)
+
+        obs_points = numpy.random.randint(-10, 10, 100)
+        below_points = [obs_points[:25]]
+        above_points = [obs_points[25:]]
+        points = tpe.sample_one_dimension(dim1, 1,
+                                          below_points, above_points, tpe._sample_int_point)
+        assert len(points) == 1
```

Comment:
It would be good to test the values as well. That they are integers in the correct interval and correspond roughly to the best region given below_points and above_points.